### ISSUE TITLE
Question about table structure

### ISSUE THREAD

**Issue by rhux** [pgRouting/osm2pgrouting#172]
I'm new to routing and osm in general so I don't know if I've missed something.  But what is the difference between source and source_osm, and target and target_osm in the ways table?  

when doing pgr_analyzeGraph on the ways table should it use source or source_osm as the source column?

Thank you.

**Comment by cvvergara** [comment]
As you are new to pgRouting I suggest you to read the latest [workshop](http://workshop.pgrouting.org)


In osm you have a way:

```
  <way id="4010204" version="24" timestamp="2014-12-23T16:46:06Z" changeset="27654675" uid="39774" user="MKnight">
    <nd ref="21185804"/>
    <nd ref="1426288008"/>
    <nd ref="1426288010"/>
    <nd ref="254127933"/>
    <nd ref="21185803"/>
    <nd ref="1426288014"/>
    <nd ref="21185800"/>
    <tag k="highway" v="motorway"/>
    <tag k="lanes" v="2"/>
    <tag k="lit" v="no"/>
    <tag k="maxspeed" v="100"/>
    <tag k="oneway" v="yes"/>
    <tag k="ref" v="A 59"/>
  </way>
```

that will be converted to many "ways", because some nodes are intersections. in this exercise I suppose all nodes are intersections:

id | osm_id | source | source_osm | target | target_osm
-----|-------|------|--------|----|----------
1 | 4010204| 1 | 21185804 | 2 | 1426288008 |
2 | 4010204| 2 | 1426288008 | 3 | 1426288010 |
3 | 4010204| 3 | 1426288010 |4 | 254127933 |
4 | 4010204| 4 | 254127933 |5| 21185803 |
5 | 4010204| 5| 21185803 |6| 1426288014 |
6 | 4010204| 6 | 1426288014 |7| 21185800

routing from vertex 1 (21185804 in osm terms) to vertex 7 (21185800 in osm terms)
```
`SELECT * from pgr_dijkstra(
  'SELECT id, source_osm AS source, target_osm AS target, cost, reverse_cost FROM ways'
  21185804, 21185800)
```
is equivalent to:
```
`SELECT * from pgr_dijkstra(
  'SELECT id, source, target, cost, reverse_cost FROM ways'
  1, 7)
```


I am going to add manually this additional way, because my application needs it:  maybe is a top secret private way in my top secret compound, so that secrecy can not be added to the public open free data on OSM
(not showing other column values, that I have to manually put values also,)

id | osm_id | source | source_osm | target | target_osm
-----|-------|------|--------|----|----------
7 | | 7 | 21185804 | 8 |  |

- edge 7 does not exist on osm, so osm_id has no value.
- in the vertex table, vertex 8 had to also be added, but that data does not come from osm so there is no corresponding osm_id valid number for that vertex in the ways_vertices_pgr table
- target_osm has no value because vertex 8 does not come from osm data

routing from vertex 1 to vertex 8:

```
`SELECT * from pgr_dijkstra(
  'SELECT id, source, target, cost, reverse_cost FROM ways'
  1, 8)
```

and can not using the osm columns `source_osm, target_osm`

So basically those columns  that hold identifiers `_osm` purpose is information for knowing the original values in OSM where the information comes from.

**Comment by rhux** [comment]
Thank you, that makes a lot of sense.  That is sort of what I had guessed was the case, but wanted to make sure before continuing with my assumptions.

Again thank you.
